lib/services/profiles-api-service.js

```diff
@@ -75,37 +75,23 @@ function profileApiServiceFactory(
         return _.flattenDeep([macs]);
     };
 
-    ProfileApiService.prototype.setLookup = function(req) {
+    ProfileApiService.prototype.setLookup = function(req, res) {
         var query = req.swagger ? req.swagger.query : req.query;
-        if (query.mac && query.ip) {
-            return waterline.nodes.findByIdentifier(this.getMacs(query.mac))
-            .then(function (node) {
-                if (_.isUndefined(node)) {
-                    return lookupService.setIpAddress(
-                        query.ip,
-                        query.mac
-                    );
-                }
-            }).then(function (node) {
-                if ( node && req.get(Constants.HttpHeaders.ApiProxyIp) ) {
-                    var proxy =  'http://%s:%s'.format(
-                        req.get(Constants.HttpHeaders.ApiProxyIp),
-                        req.get(Constants.HttpHeaders.ApiProxyPort)
-                    );
-                    return waterline.lookups.upsertProxyToMacAddress(proxy, query.mac);
-                } else {
-                    return node;
-                }
-            });
-        }
 
         if (query.macs && query.ips) {
             var macAddresses = _.flattenDeep([query.macs]);
             var ipAddresses = _.flattenDeep([query.ips]);
             return Promise.map(ipAddresses, function(ip, index) {
-                if (ip && macAddresses[index]) {
+                if (ip && macAddresses[index] && (ip === res.locals.ipAddress)) {
                     return lookupService.setIpAddress(ip, macAddresses[index]);
                 }
+                if (req.get(Constants.HttpHeaders.ApiProxyIp) && macAddresses[index]) {
```

Comment:
Two problems here:
- seems need to call upsertProxyToMacAddress only if ip is valid and (ip==res.locals.ipAddress) as the line 85 has screen the invalid ip/mac combination, so even you call upsertProxyTomacAddress, it will fail.
- The two promises `lookupService.setIpAddress` and ` waterline.lookups.upsertProxyToMacAddress` need be included in a `Promise.each`.